### DISCUSSION TITLE
ApplicationContext does not have user Blazor Server

### DISCUSSION THREAD

**Discussion by CProsRodneyWorkman** [MarimerLLC/csla#3008] [Questions]
**Describe the bug**
I have a service that calls Portal Create and it found an authenticated user from the HttpContextAccessor moments later the BusinessBase Create method ApplicationContext does not have a user.

**Version and Platform**
Csla.Blazor 6.0.0
OS: Windows, Linux, iOS, Android, etc.
Platform: Blazor Server Side

**Code that Fails**

        [Create, RunLocal]
        private void Create()
        {
            if (this.ApplicationContext.Principal.Identity==null)
                throw new NullReferenceException("Application User is missing");

            Id = Guid.NewGuid();
            base.Child_Create();
        }

**Stack Trace or Exception Detail**
Provide a stack trace, or at least detailed information about the exception.

**Additional context**
In my Login.cshtml.cs I set the Principal
            await HttpContext.SignInAsync(
                    CookieAuthenticationDefaults.AuthenticationScheme,
                    principal,
                    authProperties);

            ApplicationContext.Principal = principal;

        public AnimalEdit CreateAnimal()
        {

            var identity = HttpContextAccessor.HttpContext.User.Identity;
            if (identity.IsAuthenticated)
                return AnimalEditPortal.Create();
            else
                return null;
        }
[Program.zip](https://github.com/MarimerLLC/csla/files/8981766/Program.zip)

**Comment by rockfordlhotka** [comment]
This sounds like a bug, I'll have to look into this. Thank you for reporting!

**Comment by rockfordlhotka** [comment]
@CProsRodneyWorkman I should confirm - this is with the data portal using `LocalProxy` right?

**Comment by CProsRodneyWorkman** [comment]
Here is my configuration.  
IServiceCollection c = builder.Services.AddCsla(o => o
  .DataPortal(dpo => dpo
    .AddServerSideDataPortal()
    .UseLocalProxy()));

**Comment by rockfordlhotka** [comment]
Oh, looking at your config, I'm pretty sure the problem is that you haven't brought in the services necessary for aspnetcore or server-side Blazor:

```c#
builder.Services.AddCsla(o => o
  .AddAspNetCore()
  .AddServerSideBlazor()
  .DataPortal(dpo => dpo
    .AddServerSideDataPortal()
    .UseLocalProxy()));
```

or really just this:

```c#
builder.Services.AddCsla(o => o
  .AddAspNetCore()
  .AddServerSideBlazor());
```

because the data portal defaults to the configuration in that code.

All the services necessary to properly manage the user principal in aspnetcore and server-side Blazor (which is an intricate process) are enabled by the `AddAspNetCore` and `AddServerSideBlazor` methods.

**Comment by rockfordlhotka** [comment]
I added a new [BlazorServerExample](https://github.com/MarimerLLC/csla/tree/main/Samples/BlazorServerExample) project to the repo that demostrates the basics.
